include/mbedtls/cipher.h

```diff
@@ -480,7 +480,7 @@ static inline size_t mbedtls_cipher_info_get_key_bitlen(
     if (info == NULL) {
         return 0;
     } else {
-        return info->MBEDTLS_PRIVATE(key_bitlen) << MBEDTLS_KEY_BITLEN_SHIFT;
+        return ((size_t) info->MBEDTLS_PRIVATE(key_bitlen)) << MBEDTLS_KEY_BITLEN_SHIFT;
```

Comment:
What a weird warning: `key_bitlen` is a 4-bit bitfield, and it's shifted left by 6, so the result must fit in 10 bits, which is guaranteed to fit both into `int` (which the bit-field is promoted into) and `size_t` (which the `int` value resulting from the shift is converted to). This is one of the few cases where the compiler can know that the left shift will not cause an overflow!